src/app/item-page/simple/field-components/clarin-generic-item-field/clarin-generic-item-field.component.ts

```diff
@@ -40,15 +44,74 @@ export class ClarinGenericItemFieldComponent implements OnInit {
    */
   @Input() label: string;
 
+  /**
+   * UI URL loaded from the server.
+   */
+  baseUrl = '';
+
   // tslint:disable-next-line:no-empty
-  constructor() { }
+  constructor(protected dsoNameService: DSONameService,
+              protected configurationService: ConfigurationDataService) { }
 
   // tslint:disable-next-line:no-empty
-  ngOnInit(): void {
+  async ngOnInit(): Promise<void> {
+    await this.assignBaseUrl();
   }
 
+  /**
+   * If the metadata fields has some metadata value - show nothing if the field do not have any value.
+   */
   public hasMetadataValue() {
     return isNotUndefined(this.item.firstMetadataValue(this.fields));
   }
 
+  /**
+   * Return current metadata value. The metadata field could have more metadata values, often the metadata
+   * field has only one metadata value - index is 0, but sometimes it has more values e.g. `Author`.
+   * @param index
+   */
+  public getLinkToSearch(index, value = '') {
+    let metadataValue = 'Error: value is empty';
+    if (isEmpty(value)) {
+      // Get metadata value from the Item's metadata field
+      metadataValue = this.getMetadataValue(index);
+    } else {
+      // The metadata value is passed from the parameter.
+      metadataValue = value;
+    }
+
+    const searchType = convertMetadataFieldIntoSearchType(this.fields);
+    return this.baseUrl + '/search/objects?f.' + searchType + '=' + metadataValue + ',equals';
+  }
+
+  /**
+   * If the metadata field has more than 1 value return the value based on the index.
+   * @param index of the metadata value
+   */
+  public getMetadataValue(index) {
+    let metadataValue = '';
+    if (index === 0) {
+      // Return first metadata value.
+      metadataValue = this.item.firstMetadataValue(this.fields);
```

Comment:
please, avoid using `if` this way
if (index === 0) return this.item.first.......

Reply:
fixed